akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -988,10 +991,18 @@ private[akka] class Shard(
         log.debug("Started entity [{}] with entity id [{}] in shard [{}]", a, id, shardId)
         entities.addEntity(id, a)
         touchLastMessageTimestamp(id)
+        entityCreated(id)
         a
     }
   }
 
+  /**
+   * Called when an entity has been created. Returning the number
+   * of active entities.
+   */
+  @InternalStableApi
+  def entityCreated(@unused id: EntityId): Int = entities.activeEntities().size
```

Comment:
It needs to return the number because cinnamon wants that number? Shouldn't it rather be `Unit` and cinnamon calls whatever it wants then?

Reply:
That's right. Doing it this way means we don't need to interact with the Entities class and stub it out there

Reply:
Fair enough.